src/tiler/healthCheck.js

```diff
@@ -0,0 +1,106 @@
+var Step = require('windshaft/node_modules/step');
+var Pg = require('windshaft/node_modules/cartodb-psql/node_modules/pg');
+var Redis = require('windshaft/node_modules/redis-mpool/node_modules/redis');
+
+/**
+ * Attempts to connect to a PostgreSQL database, providing the
+ * callback with a success or failure boolean.
+ * @param {string} host
+ * @param {string} port
+ * @param {string} user
+ * @param {string} password
+ * @param {string} database
+ * @param {function} callback
+ */
+function checkPostgres(host, port, user, password, database, callback) {
+  var conString = ['postgres://', user, ':', password, '@', host, '/', database].join('');
+  var client = new Pg.Client(conString);
+
+  client.connect(function(err) {
+    client.end();
+    callback(!err);
+  });
+}
+
+/**
+ * Attempts to connect to a Redis data store, providing the
+ * callback with a success or failure boolean.
+ * @param {string} host
+ * @param {string} port
+ * @param {function} callback
+ */
+function checkRedis(host, port, callback) {
+  var client = Redis.createClient(port, host);
+  var wrapper = function(status) {
+    client.quit();
+    callback(status);
+  };
+
+  client.on('connect', function() {
+    wrapper(true);
+  });
+
+  client.on('error', function() {
+    wrapper(false);
+  });
+}
+
+/**
+ * Makes use of the PostgreSQL and Redis checks above to return
+ * a JSON object representing the health of both data stores:
+ *
+ * {
+ *   "databases": [
+ *     {
+ *       "default": {
+ *         "ok": true
+ *       }
+ *    }
+ *   ],
+ *   "caches": [
+ *     {
+ *       "default": {
+ *         "ok": true
+ *       }
+ *     }
+ *   ]
+ * }
+ *
+ * @param {object} config - The configuration object passed to Windshaft.Server()
+ */
+module.exports = function createHealthCheckHandler(config) {
+  var pgConf = config.grainstore.datasource,
+
+    pgHost = pgConf.host,
+    pgPort = pgConf.port,
+    pgUser = pgConf.user,
+    pgPassword = pgConf.password,
+    pgDatabase = pgConf.dbname,
+
+    redisHost = config.redis.host,
+    redisPort = config.redis.port;
+
+  return function(req, res) {
+    var status = {};
+
+    Step(
+      function database() {
+        checkPostgres(pgHost, pgPort, pgUser, pgPassword, pgDatabase, this);
+      },
+      function cache(databaseStatus) {
+        checkRedis(redisHost, redisPort, function(redisStatus) {
+          var response = {};
+          var statusCode = 503;
+
+          response.databases = [{default: {ok: databaseStatus}}];
+          response.caches = [{default: {ok: redisStatus}}];
+
+          if (databaseStatus && redisStatus) {
+            statusCode = 200;
+          }
+
+          res.send(response, statusCode);
+        });
```

Comment:
Addressed in https://github.com/WikiWatershed/model-my-watershed/commit/5eedffb99b529ad90823b4bdf38fdffa1dbdcfc1.